skfem/assembly/basis/interior_basis.py

```diff
@@ -154,6 +154,10 @@ def probes(self, x: ndarray) -> coo_matrix:
             shape=(x.shape[1], self.N),
         )
 
+    def point_source(self, x: ndarray) -> ndarray:
+        """Return right-hand side vector for unit source at `x`"""
```

Comment:
Is this docstring true for other than P1 Lagrange?

Reply:
Hmm, I think so, in the Galerkin sense. Initially I reread Jan's point-source function in #561 and was reminded of what we'd done for the probe for Nico. Then i thought that at the continous variational level, interpolation is the inner product of a finite element function with a Dirac delta and a point source is the inner product of a Dirac delta with a test function. Those are adjoints and on discretization they're transposes. That line of thoight doesn't make any reference to the txpe of basis.

Similarly for `skfem.models.poisson.unit_load` and the functional that integrates, replacing the Dirac delta with the constant 1.

Reply:
I don't have a reference for this. I couldn't find one in a quick look. I remember that for years users were asking about point sources on other finite element msiling lists. Somtimes the discussion suggested mollification: convolving the point source with a Gaussian kernel, but I didn't like that so much as it still didn't seem well suited to the quadrature.  Then FreeFEM added one recently; I refrained from looking at  their implementation as it's GPL.

Reply:
Possible hole in the argument: the argument shows that the transpose property holds but it only follows from that that the function returns the Galerkin point source vector if the probes function works correctly. I didn't look into whether that works for all kinds of bases. Does it? I assumed that it did.

Reply:
A couple of brief references to point sources in some of the older FEM books on my self:
* Segerlind, L. J. (1976). *Applied Finite Element Analysis.* New York: Wiley, &sect;8.6 &lsquo;Point sources&rsquo;
* Hughes, T. J. R. (2000). *The Finite Element Method.* Mineola, New York: Dover, exercise 3.11.2(*b*)

Unfortunately Hughes doesn't give the solution to the exercise (delta function loading on a one-dimensional, quadratic, three-node element), but I think it's looking the similar to my pair of [linear two-node elements](https://github.com/kinnala/scikit-fem/issues/629#issuecomment-823701775).

Segerlind notes:

> The integral of a function multiplied by an impulse function, however, is simply the function evaluated at <i>X</i><sub>0</sub> and <i>Y</i><sub>0</sub>.

which is elementary and the subsequent example is only for `ElementTriP1`.

I'm yet to turn up anything on nonlagrangian elements.